### PR TITLE
fix MPP-2494: check References header for original message id

### LINKED ISSUE #1988: Second reply is not delivered.
Steps to reproduce:

1. Send an email to an alias
2. Receive the email, and send a reply
3. Send a second reply

The first reply is delivered, the second is not.

### PR DESCRIPTION
**Pull request by groovecoder**
This PR fixes #1988 MPP-2494.

How to test:
Original bug STR: https://github.com/mozilla/fx-private-relay/issues/1988#issue-1245761814

- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: emails/tests/fixtures/dmarc_failed_email_sns_body.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
...\":\"<CA+J4FJFjMP7djF0ph8f_pPQg6FvCu_vWuUbG2QxtjPtikvYWXw@mail.gmail.com>\"},{\"name\":\"Subject\",\"value\":\"Re: Test Mozilla User New Domain Address\"},{\"name\":\"To\",\"value\":\"sender@test.com\"},{\"name\":\"Content-Type\",\"value\":\"multipart/alternative; boundary=\\\"0000000000003d386c05ccffc614\\\"\"}],\"commonHeaders\":{\"returnPath\":\"source@sender.com\",\"from\":[\"Sender Name <source@sender.com>\"],\"date\":\"Mon, 27 Sep 2021 15:16:26 -0500\",\"to\":[\"sender@test.com\"],\"messageId\":\"<CA+J4FJFjMP7djF0ph8f_pPQg6FvCu_vWuUbG2QxtjPtikvYWXw@mail.gmail.com>\",\"subject\":\"Re: Test Mozilla User New Domain Address\"}},\"receipt\":{\"timestamp\":\"2021-09-27T20:16:39.558Z\",\"processingTimeMillis\":1086,\"recipients\":[\"sender@test.com\"],\"spamVerdict\":{\"status\":\"PASS\"},\"virusVerdict\":{\"status\":\"PASS\"},\"spfVerdict\":{\"status\":\"FAIL\"},\"dkimVerdict\":{\"status\":\"PASS\"},\"dmarcVerdict\":{\"status\":\"FAIL\"},\"dmarcPolicy\":\"reject\",\"action\":{\"type\":\"S3\",\"topicArn\":\"arn:aws:sns:us-east-1:927034868273:fxprivaterelay-SES-processor-topic\",\"bucketName\":\"test-bucket\",\"objectKeyPrefix\":\"/emails\",\"objectKey\":\"/emails/objectkey123\"}}}",
```
Comment (jwhitlock) [inline emails/tests/fixtures/dmarc_failed_email_sns_body.json:6]: How did you make the changes to these fixtures? Is it from a live email or manual guesses?

File: emails/tests/fixtures/s3_stored_replies_email_sns_body.json [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
   "Timestamp": "2021-09-27T20:16:40.657Z",
   "SignatureVersion": "1",
   "Signature": "gheTllFhtCKa9D61BfHZ7NCQyDNo5HzlS7zAmOsgBnkvlqDh3KJE7ye4y3tlyHDQU0fS1UDRP6oii0Q49iXz8klv3FskcbzpX563y7I5+Xvbzu2AelOmiaAgiBGTiqwGD6XSHfdLUeo3jcVftjzkCwxkyNAM5R7+wrvnfm3/Kbr0SvLjTysHfZilXHujFV13E2uYxdA43Sx2K63Rhbqg1jPIPyEIZXS2o8remdrd0zlyUIQvGwA+9vvzzjOftHA4iZ9UY84NhXoV22i9FMaijjE2pqJX4dn/yQN+T84LZhsHpjuvdUKVIP1aozSOoL9hvbTAiLdCLtJOrG0NWG/G1A==",
   "SigningCertURL": "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-7ff5318490ec183fbaddaa2a969abfda.pem",
   "UnsubscribeURL": "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:927034868273:fxprivaterelay-SES-processor-topic:2bf6ca66-7f5a-4bf7-a53a-43b08c3a2676"
-}
\ No newline at end of file
+}
```
Comment (jwhitlock) [inline emails/tests/fixtures/s3_stored_replies_email_sns_body.json:12]: I will always ➕ adding a newline at the end of docs.

File: emails/views.py [hunk 3]
```diff
@@ -581,10 +581,21 @@ def _get_keys_from_headers(headers):
...
+                message_id_bytes = get_message_id_bytes(message_id)
+                lookup_key, encryption_key = derive_reply_keys(message_id_bytes)
+                try:
+                    # FIXME: calling code is likely to duplicate this query
+                    _get_reply_record_from_lookup_key(lookup_key)
+                    return lookup_key, encryption_key
+                except Reply.DoesNotExist:
+                    pass
+            raise Reply.DoesNotExist
+    raise ReplyHeadersNotFound
```
Comment (jwhitlock) [inline emails/views.py:598]: Should we still call `incr_if_enabled("mail_to_replies_without_in_reply_to", 1)`?

**Review by jwhitlock — APPROVED** [review]
Thanks @groovecoder! This change worked for me, I was able to send a second reply.

I have a few questions. Feel free to merge after changes or re-request a review.

File: emails/tests/fixtures/dmarc_failed_email_sns_body.json (hunk 1 above)
Comment (groovecoder) [reply emails/tests/fixtures/dmarc_failed_email_sns_body.json:6]: I made manual changes.

File: emails/views.py (hunk 3 above)
Comment (groovecoder) [reply emails/views.py:598]: Good call. I changed it to `mail_to_replies_without_reply_headers`.

**Pull request merged**